packages/eslint-plugin/src/rules/consistent-generic-constructors.ts

```diff
@@ -29,10 +34,24 @@ export default createRule<Options, MessageIds>({
         description: 'Which constructor call syntax to prefer.',
         enum: ['type-annotation', 'constructor'],
       },
+      {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          ignore: {
+            type: 'array',
+            description:
+              'A list of constructor names to ignore when enforcing the rule.',
```

Comment:
IMO if we do go with a list then we'd want to use [TypeOrValueSpecifier](https://typescript-eslint.io/packages/type-utils/type-or-value-specifier). Otherwise it'll have the same issue as https://github.com/typescript-eslint/typescript-eslint/pull/10477#discussion_r1885099562 with coincidentally matching names.

But, I'm not sure this is the right step. I started a thread later in this file about it.